hack/check-api-diff.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+#
+# Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+API_DIFF="hack/tools/bin/go-apidiff"
+
+cleanup_output () { rm -fr "output.txt" "result.txt"; }
+
+retval=0
+temp=0
+
+echo "invoking: go-apidiff master --print-compatible --repo-path=."
```

Comment:
Maybe the base ref (`master`) should be configurable, so that we can later on run it on release branches as well

Reply:
For now, let's check on master for some time then in the future we can make it configurable.